### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by souporserious** [souporserious/react-media-player#4]
Will update soon..

Inspiration: http://terrillthompson.com/blog/32

**Comment by zalmoxisus** [comment]
Supporting events (like `onMute`, `onPause`, `OnStart`, `onSeek`) would be useful.

**Comment by souporserious** [comment]
In favor of keeping the library small and not having to code that for each vendor I figured it could be achieved in `componentDidUpdate` with something like the following:

```
componentDidUpdate(lastProps) {
  if (lastProps.isMuted !== this.props.isMuted) {
    // handle onMute
  }
}
```

does that work for you?

**Comment by zalmoxisus** [comment]
That's better indeed. Any way to watch when `seekTo` is called, at least from inside the `SeekBar` (to use instead of `onSeek`)?

**Comment by souporserious** [comment]
I think `currentTime` would be your best bet.

```
componentDidUpdate(lastProps) {
  if (lastProps.currentTime !== this.props.currentTime) {
    // handle onSeek
  }
}
```
Please let me know if that works or not.

**Comment by zalmoxisus** [comment]
Thanks! I thought `currentTime` is synchronised during the video playing, but it appears to be as `startingTime`, so it will work for sure.

**Comment by sascha1337** [comment]
yeah! looking forward to see soundcloud integration.
thx 4 this lib

**Comment by souporserious** [comment]
@zalmoxisus implemented actual callbacks in the latest release 🎉  https://github.com/souporserious/react-media-player#onplay-proptypesfunc

**Comment by oyeanuj** [comment]
@souporserious Any plans to add demo link (github-pages, or codepen)? It would be great to be able to see the functionality in action from the repo itself!

**Comment by souporserious** [comment]
Yes, I totally agree. Just been slammed with work lately. I will have live demos up soon!

**Comment by talolard** [comment]
Hey,
This is super useful. I'd love to contribute support for more audio file formats but have no idea if that's within reach. Could someone point me to the right docs about html5 audio formats and I'll see what I can do? 
Thanks

**Comment by souporserious** [comment]
Thanks @talolard. I'd love to help you get onboard. If you can give me a week or two, I'll have better documentation for everything so contributing will be easier. Are there any file formats you found that don't work currently that you want?

**Comment by florimm** [comment]
Any plans to upgrade to react 16.3 with new Context API?

**Comment by souporserious** [comment]
Yes! Sorry this has been a little stale. I plan to do an update and finish [this](https://github.com/souporserious/react-media-player/pull/25) as soon as I get some free time 😅.

**Comment by shahankit** [comment]
Hi @souporserious, any updates when the new context api changes will be released?

**Comment by souporserious** [comment]
Sorry, no solid timeline right now 😞. I'm wrapping up a few other projects that I've been consumed with, but once they are done I'll hop on here and get it done.
